### PR TITLE
Return statements for modules xapi, xmpp and yumpkg

### PR DESCRIPTION
**Pull request by michaelcoyote**
Added return statements for **virtual** on modules xapi, xmpp and yumpkg

### CODE REVIEW

File: salt/modules/yumpkg.py [hunk 1]
```diff
@@ -59,18 +59,18 @@ def __virtual__():
     Confine this module to yum based systems
     '''
     if __opts__.get('yum_provider') == 'yumpkg_api':
-        return False
+        return (False, "Module yumpkg: yumpkg_api provider not avalible")
```
Comment (techhat) [inline salt/modules/yumpkg.py]: `available`

**Comment by michaelcoyote** [comment]
@techhat thanks.  Updated w/ correct spelling.

**Comment by cachedout** [comment]
Hi @michaelcoyote Thanks for these. I do think that going forward we could be a little more explicit about giving the user some actionable information about how to fix the problem. For example, we could show them _which_ python packages need to be installed to enable the given functionality. Just a thought. :] Thanks for all your help here though. It's very much appreciated!

**Comment by michaelcoyote** [comment]
@cachedout No problem.  I wasn't sure how much I should try to change in one bite.  Some of the virtual function blocks do the checks in a separate function and don't actually return the reason for failure, so it it does make it harder to pass the reason out from virtual. 
To fix it, I'd need to make the changes on that next function down so that it properly passes the reason up to virtual and then pass the message out from virtual.  
The other option would be to refactor the virtual to move the logic from the embedded function to virtual and do it's checking from within virtual.  This would only be an option if there were no other calls to the embedded function.  

If it's useful I can start getting input from IRC when I run across these.  It's easy enough for me to do.

**Pull request merged**